packages/cli/src/ui/contexts/KeypressContext.tsx

```diff
@@ -603,6 +603,83 @@ export function KeypressProvider({
       if (key.name === 'return' && key.sequence === `${ESC}\r`) {
         key.meta = true;
       }
+
+      // Map common Alt+letter combinations that produce Unicode characters on macOS
+      // This handles terminals that don't send ESC sequences (like iTerm2 default)
+      const altUnicodeMap: {
+        [unicode: string]: { key: string; shift: boolean };
+      } = {
+        // Alt + lowercase letters
+        å: { key: 'a', shift: false },
+        '∫': { key: 'b', shift: false },
+        ç: { key: 'c', shift: false },
+        '∂': { key: 'd', shift: false },
+        ƒ: { key: 'f', shift: false },
+        '©': { key: 'g', shift: false },
+        '˙': { key: 'h', shift: false },
+        ˆ: { key: 'i', shift: false },
+        '∆': { key: 'j', shift: false },
+        '˚': { key: 'k', shift: false },
+        '¬': { key: 'l', shift: false },
+        µ: { key: 'm', shift: false },
+        ø: { key: 'o', shift: false },
+        π: { key: 'p', shift: false },
+        œ: { key: 'q', shift: false },
+        '®': { key: 'r', shift: false },
+        ß: { key: 's', shift: false },
+        '†': { key: 't', shift: false },
+        '√': { key: 'v', shift: false },
+        '∑': { key: 'w', shift: false },
+        '≈': { key: 'x', shift: false },
+        '¥': { key: 'y', shift: false },
+        Ω: { key: 'z', shift: false },
+        // Alt + uppercase letters (Shift+Alt)
+        Å: { key: 'a', shift: true },
+        Ø: { key: 'o', shift: true },
+        '∏': { key: 'p', shift: true },
+        Œ: { key: 'q', shift: true },
+      };
+
+      const unicodeChar = key.sequence || key.name;
+      if (
+        unicodeChar &&
+        unicodeChar.length === 1 &&
+        altUnicodeMap[unicodeChar]
+      ) {
+        const mapped = altUnicodeMap[unicodeChar];
+        broadcast({
+          name: mapped.key,
+          ctrl: false,
+          meta: true,
+          shift: mapped.shift,
+          paste: isPaste,
+          sequence: key.sequence,
+        });
```

Comment:
all these keypress context changes need tests.ideally they would be separate pr so we can review separate from the UX discussions for the markdown display.